core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/QueryOptimizerFunctionalInterface.java

```diff
@@ -0,0 +1,18 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.query.algebra.evaluation;
+
+import org.eclipse.rdf4j.query.algebra.evaluation.impl.EvaluationStatistics;
+
+public interface QueryOptimizerFunctionalInterface {
```

Comment:
I don't understand the point of this functional interface to be honest. It does not seem to make the code easier or more compact.

Reply:
The reason is that none of the variables are available when creating a EvaluationStrategyFactory, so they need to be passed at a later point.